tests/chainer_tests/iterators_tests/test_serial_iterator.py

```diff
@@ -308,4 +329,68 @@ def test_iterator_serialize_backward_compat(self):
         self.assertAlmostEqual(it.previous_epoch_detail, 4 / 6)
 
 
+class TestSerialIteratorOrderSamplerEpochSize(unittest.TestCase):
+
+    def setUp(self):
+        def order_sampler(order, cur_pos):
+            return numpy.repeat(numpy.arange(3), 2)
+        self.options = {'shuffle': False, 'order_sampler': order_sampler}
+
+    def test_iterator_repeat(self):
+        dataset = [1, 2, 3]
+        it = iterators.SerialIterator(dataset, 2, **self.options)
+        for i in range(3):
+            self.assertEqual(it.epoch, i)
+            self.assertAlmostEqual(it.epoch_detail, i + 0 / 6)
+            if i == 0:
+                self.assertIsNone(it.previous_epoch_detail)
+            else:
+                self.assertAlmostEqual(it.previous_epoch_detail, i - 2 / 6)
+            batch1 = it.next()
+            self.assertEqual(len(batch1), 2)
+            self.assertIsInstance(batch1, list)
+            self.assertFalse(it.is_new_epoch)
+            self.assertAlmostEqual(it.epoch_detail, i + 2 / 6)
+            self.assertAlmostEqual(it.previous_epoch_detail, i + 0 / 6)
+            batch2 = it.next()
+            self.assertEqual(len(batch2), 2)
+            self.assertIsInstance(batch2, list)
+            self.assertFalse(it.is_new_epoch)
+            self.assertAlmostEqual(it.epoch_detail, i + 4 / 6)
+            self.assertAlmostEqual(it.previous_epoch_detail, i + 2 / 6)
+            batch3 = it.next()
+            self.assertEqual(len(batch3), 2)
+            self.assertIsInstance(batch3, list)
+            self.assertTrue(it.is_new_epoch)
+            self.assertAlmostEqual(it.epoch_detail, i + 6 / 6)
+            self.assertAlmostEqual(it.previous_epoch_detail, i + 4 / 6)
+
+            self.assertEqual(
+                sorted(batch1 + batch2 + batch3), [1, 1, 2, 2, 3, 3])
+
+
+class InvalidOrderSampler(object):
+
+    def __init__(self):
+        self.n_call = 0
+
+    def __call__(self, _order, _):
+        order = numpy.arange(len(_order) - self.n_call)
+        self.n_call += 1
+        return order
+
+
+class TestSerialIteratorInvalidOrderSampler(unittest.TestCase):
+
+    def test_invalid_order_sampler(self):
+        dataset = [1, 2, 3, 4, 5, 6]
+
+        with self.assertRaises(ValueError):
+            it = iterators.SerialIterator(
+                dataset, 6, shuffle=False,
+                order_sampler=InvalidOrderSampler())
+            it.next()
+            it.next()
```

Comment:
I don't think this line is executed/needed since `_order` is first set in the constructor (`reset()`).

Reply:
Thank you.